csaf_2.1/json_schema/csaf_json_schema.json

```diff
@@ -1201,6 +1201,57 @@
               }
             }
           },
+          "metrics": {
+            "title": "List of metrics",
+            "description": "Contains metric objects for the current vulnerability.",
+            "type": "array",
+            "minItems": 1,
+            "uniqueItems": true,
+            "items": {
+              "title": "metric",
+              "description": "Contains all metadata about the metric including products it applies to and the source and the content itself.",
+              "type": "object",
+              "required": [
+                "content",
+                "products"
+              ],
+              "properties": {
+                "content": {
+                  "title": "Content",
+                  "description": "Specifies information about (at least one) metric or score for the given products regarding the current vulnerability.",
+                  "type": "object",
+                  "minProperties": 1,
+                  "properties": {
+                    "cvss_v2": {
+                      "$ref": "https://www.first.org/cvss/cvss-v2.0.json"
+                    },
+                    "cvss_v3": {
+                      "oneOf": [
+                        {
+                          "$ref": "https://www.first.org/cvss/cvss-v3.0.json"
+                        },
+                        {
+                          "$ref": "https://www.first.org/cvss/cvss-v3.1.json"
+                        }
+                      ]
+                    },
+                    "cvss_v4": {
+                      "$ref": "https://www.first.org/cvss/cvss-v4.0.json"
+                    }
+                  }
+                },
+                "products": {
+                  "$ref": "#/$defs/products_t"
+                },
+                "source": {
+                  "title": "Source",
+                  "description": "Contains the URL of the source that originally determined the metric.",
+                  "type": "string",
+                  "format": "uri"
```

Comment:
Does it make sense to restrict this to URIs? The initial feature request notes "...add an optional source (or reference)", which makes me believe that setting this to an arbitrary text value is also desired.

Reply:
I am not using this currently, but URLs might be more meaningful for consumers than general text, no? Something more like an identifier that one could build hard relational anlaysis on, kind of.

Reply:
@jaccoNCSCNL, any preference from you since you originally requested this feature?

Reply:
As our goal should be automation, I'm quite hesitant to add generic text... As a source does not fall out of the sky, I think a URL is a reasonable choice. (It would have also been a URL if we choose to implement it via `references`).